packages/constants/src/constants.ts

```diff
@@ -11,31 +11,65 @@ export const HIDE_LOADING = 'HIDE_LOADING';
 export const HIDE_ALL_PAGE_NOTIFICATIONS = 'HIDE_ALL_PAGE_NOTIFICATIONS';
 
 // Notifications
-export type TAppNotificationDomain = 'global' | 'page' | 'side';
-export const DOMAINS: {
-  [key: string]: TAppNotificationDomain;
+export const NOTIFICATION_DOMAINS: {
+  GLOBAL: 'global';
+  PAGE: 'page';
+  SIDE: 'side';
 } = {
   GLOBAL: 'global',
   PAGE: 'page',
   SIDE: 'side',
 };
+export const NOTIFICATION_KINDS_SIDE: {
+  error: 'error';
+  warning: 'warning';
+  info: 'info';
+  success: 'success';
+} = {
+  error: 'error',
+  warning: 'warning',
+  info: 'info',
+  success: 'success',
+};
+export const NOTIFICATION_KINDS_GLOBAL: {
+  error: 'error';
+  warning: 'warning';
+  info: 'info';
+  success: 'success';
+  'unexpected-error': 'unexpected-error';
+} = {
+  error: 'error',
+  warning: 'warning',
+  info: 'info',
+  success: 'success',
+  'unexpected-error': 'unexpected-error',
+};
+export const NOTIFICATION_KINDS_PAGE: {
+  error: 'error';
+  warning: 'warning';
+  info: 'info';
+  success: 'success';
+  'unexpected-error': 'unexpected-error';
+  'api-error': 'api-error';
+} = {
+  error: 'error',
+  warning: 'warning',
+  info: 'info',
+  success: 'success',
+  'unexpected-error': 'unexpected-error',
+  'api-error': 'api-error',
+};
+export type TAppNotificationDomain = (typeof NOTIFICATION_DOMAINS)[keyof typeof NOTIFICATION_DOMAINS];
```

Comment:
This "weird" syntax basically extracts the object values as union types.

See https://stackoverflow.com/a/50044130